### ISSUE TITLE
Wrong address template used for order e-mails

### ISSUE THREAD

**Issue by andreaskoch** [magento/magento2#7984]
Magento CE 2.1.3 and Magento EE 2.1.2 uses the wrong address template for shipping, invoice and credit memo e-mails.

## Preconditions

- Magento 2.1.3 Community Edition with sample data or
- Magento 2.1.2 Enterprise Edition
- A Magento shop with multiple websites, stores and store views

## Steps to reproduce

1. Create customized address templates on store-view level
    - Create a second store view for the default website of the Magento sample data store
    ![Create a second store view](https://cloud.githubusercontent.com/assets/247364/21532422/4bc91dd6-cd50-11e6-9629-d49bea9ef94b.png)
    - Goto `Backend > Stores > Settings > Configuration > Customers > Customer Configuration > Address Templates`
    - Change the config scope to the newly created store view.
    - Uncheck `Use Website`
    - Customize the the address templates on **store view** level
    ![Customized address templates on store view level](https://cloud.githubusercontent.com/assets/247364/21532425/59b468a6-cd50-11e6-82ca-4a25d2647712.png)
    - Clear the caches
- Place an order on that store view
    - Process the order (Ship, Invoice, Credit Memo)
- Check the generated mails
    - order confirmation
    - shipping
    - invoice
    - credit memo

## Expected result

The address template from store view level of the placed order should be used as it is done for the order confirmation e-mail:

![Actual Result: Order Confirmation E-Mail - Correct address template from store view level is used](https://cloud.githubusercontent.com/assets/247364/21532429/672031a0-cd50-11e6-9474-4dcad4e4ca09.png)

## Actual result

The address template from the default instead of the store view scope is used for the following e-mail types:

**shipping e-mail**

![Actual result: shipping e-mail - the address template of the default scope is used](https://cloud.githubusercontent.com/assets/247364/21532448/823657c6-cd50-11e6-849f-251c4b7070dd.png)

**invoice e-mail**

![Actual result: invoice e-mail - the address template of the default scope is used](https://cloud.githubusercontent.com/assets/247364/21532453/8f576cec-cd50-11e6-907e-0947fccc9b6e.png)

**credit memo e-mail**

![Actual result: credit-memo e-mail - the address template of the default scope is used](https://cloud.githubusercontent.com/assets/247364/21532461/9b778912-cd50-11e6-98f3-c66121612628.png)

## Mitigation

The order address renderer can be mitigated by rewriting the `format` function of the `\Magento\Sales\Model\Order\Address\Renderer` class:

```php
use Magento\Customer\Model\Address\Config as AddressConfig;
use Magento\Framework\Event\ManagerInterface as EventManager;
use Magento\Sales\Model\Order\Address;

/**
 * Class Renderer used for formatting an order address
 */
class Renderer extends \Magento\Sales\Model\Order\Address\Renderer
{
    /**
     * Format address in a specific way
     *
     * @param Address $address
     * @param string $type
     * @return string|null
     */
    public function format(Address $address, $type)
    {
        // If store is not set / is admin store, something went wrong - each order address should
        // belong to specific store - set it!
        if($this->addressConfig->getStore()->getId() === '0') {
            $this->addressConfig->setStore($address->getOrder()->getStore());
        }

        $formatType = $this->addressConfig->getFormatByCode($type);
        if (!$formatType || !$formatType->getRenderer()) {
            return null;
        }
        $this->eventManager->dispatch('customer_address_format', ['type' => $formatType, 'address' => $address]);
        return $formatType->getRenderer()->renderArray($address->getData());
    }
}
```

see: https://github.com/magento/magento2/blob/develop/app/code/Magento/Sales/Model/Order/Address/Renderer.php#L49

**Comment by abarua17** [comment]
@andreaskoch This bug has been acknowledged and an internal ticket MAGETWO-63135 has been created.

**Comment by magicbunneh** [comment]
Fix for this issue was merged:
- [magento2/develop](https://github.com/magento/magento2/tree/develop) - repo/branch
- [ca2ade1cb0cb043d5ef2c4590a516a0b9547a2ed](https://github.com/magento/magento2/commit/ca2ade1cb0cb043d5ef2c4590a516a0b9547a2ed) - commit

**Comment by magento-engcom-team** [comment]
@andreaskoch, thank you for your report.
The issue is already fixed in 2.2.0
